### ISSUE TITLE
[ROCK5B]  FAT partition does not exist

### ISSUE THREAD

**Issue by dclobato** [MichaIng/DietPi#6570]
#### Required Information
- DietPi version v8.20
- https://dietpi.com/downloads/images/DietPi_ROCK5B-ARMv8-Bookworm.7z
- ROCKPI5B
- eMMC 64GB

#### Steps to reproduce
1. Download https://dietpi.com/downloads/images/DietPi_ROCK5B-ARMv8-Bookworm.7z
2. Extract img from 7z
3. Verify SHA1SUM (it does match the one on hash.txt
4. Write image on eMMC 64GB using Balena Etcher

#### Expected behaviour
- There should be a FAT partition, so we can edit dietpi.txt

#### Actual behaviour
5. There is no FAT partition

**Comment by Joulinar** [comment]
This is correct and expected as the image contains an ext4 partition only.

**Comment by Joulinar** [comment]
That might be an option

**Comment by MichaIng** [comment]
This can be also a VM on Windows, like our VirtualBox DietPi appliance.

I don't like the current situation either that even with WSL2 there is no reliable way to mount Linux filesystems on Windows. But as dpkg does not support FAT properly, having a `/boot` FAT filesystem was and it's always a hacky thing with limitations.

**Comment by dirkhh** [comment]
I have been playing with this for a while since my similar question stayed unanswered.
`/boot` on FAT does have its issues, but I think I have come up with an idea that might solve this in a clean manner.
Add an optional `/firstboot` FAT partition as second partition after the ext4 partition in the images. And have the `dietpi.txt` and `dietpi-wifi.txt` files in that partition. Make this all optional. The firstboot looks for the partition. If it is there, it mounts it, copies the .txt files over the ones in `/boot/dietpi`, then unmounts the partition and deletes it.
Now the rest of the installation can continue as before.
If the partition isn't there, nothing changes. If it exists, it becomes a lot easier for users to edit those two files (and anything else that might benefit and could be put there).
If this is soething that @MichaIng finds a reasonable idea, I'll be happy to work on it and submit a PR.
(in case it isn't obvious, yes, I tried this in a very very hackish manner and it works - but I'd need to make this clean and more generic)

**Comment by MichaIng** [comment]
That's quite a nice idea. Could be all done in https://github.com/MichaIng/DietPi/blob/dev/rootfs/var/lib/dietpi/services/fs_partition_resize.sh.

The partition could be created in [dietpi-build](https://github.com/MichaIng/DietPi/blob/dev/.build/images/dietpi-build). But one difficulty is that [dietpi-imager](https://github.com/MichaIng/DietPi/blob/dev/.build/images/dietpi-imager) currently expects the rootfs to be on the last partition, otherwise it wouldn't be able to shrink the image. And it compresses it in the same turn. So actually the only way I currently see is to do this in `dietpi-imager`, probably based on an environment variable and optionally menu option 🤔.

**Comment by MichaIng** [comment]
To avoid misunderstanding in the first place: It is not like `rpi-imager` a tool to pre-configure and flash an image, but to create a minimised image (minimised filesystem and partition sizes) from an attached drive or `img` file. Our images are generated with these tools:
- `dietpi-build` as the parent build script to `debootstrap` the rootfs, do partitioning and filesystem creation and call the other scripts.
- `dietpi-installer` to install platform/choice dependent packages, install DietPi files and configure the system, all from within the system started as (`systemd-nspawn`) container.
- `dietpi-imager` to reduce the image size back to a minimum and generate an archive from it.

There are some special cases, like the Clonezilla-based x86_64 installer images and VM appliances, but for those, the extra FAT partition has no use anyway.

**Comment by dirkhh** [comment]
Yup... reading the sources has gotten me to that point already... doing what I'm trying to do seems fairly easy, given the structure. Doing it aligned with your way of writing shell code... yeah, that might take a few days 🤣

**Comment by MichaIng** [comment]
Cleaning up some old open issues: The suggestion with a tailing FAT partition has been implemented since a while, so no problem editing (first) boot configs from non-Linux systems anymore 🙂.
